cmd/manager/main.go

```diff
@@ -0,0 +1,187 @@
+// Copyright 2018 BlueData Software, Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"github.com/bluek8s/kubedirector/pkg/observer"
+	"github.com/bluek8s/kubedirector/pkg/validator"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"os"
+	"runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	// Import all Kubernetes client auth plugins (e.g. Azure, GCP, OIDC, etc.)
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+
+	"github.com/bluek8s/kubedirector/pkg/apis"
+	"github.com/bluek8s/kubedirector/pkg/controller"
+
+	kdv1 "github.com/bluek8s/kubedirector/pkg/apis/kubedirector.bluedata.io/v1alpha1"
+	"github.com/operator-framework/operator-sdk/pkg/k8sutil"
+	"github.com/operator-framework/operator-sdk/pkg/leader"
+	"github.com/operator-framework/operator-sdk/pkg/log/zap"
+	"github.com/operator-framework/operator-sdk/pkg/metrics"
+	"github.com/operator-framework/operator-sdk/pkg/restmapper"
+	sdkVersion "github.com/operator-framework/operator-sdk/version"
+	"github.com/spf13/pflag"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/signals"
+)
+
+// Change below variables to serve metrics on different host or port.
+var (
+	metricsHost       = "0.0.0.0"
+	metricsPort int32 = 8383
+)
+var log = logf.Log.WithName("cmd")
+
+func printVersion() {
+	log.Info(fmt.Sprintf("Go Version: %s", runtime.Version()))
+	log.Info(fmt.Sprintf("Go OS/Arch: %s/%s", runtime.GOOS, runtime.GOARCH))
+	log.Info(fmt.Sprintf("Version of operator-sdk: %v", sdkVersion.Version))
+}
+
+func main() {
+	// TODO: the operator-sdk switched to a new log implementation.
+	//       should we migrate the shared logger code?
```

Comment:
Yeah. We don't have any particular attachment to logrus, and the fewer random thirdparty dependencies we have the better.

Reply:
I fixed the Makefile compile target and will migrate our code to the new log implementation.

Reply:
Regarding the new log implementation, each Reconcile request creates a new logger object that will tag all log messages with the request's namespace and name:
	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
This eliminates the need for some of the existing shared.logger code that takes the KubeDirectorCluster namespace/name and prepends them to all of the log messages. This also makes the shared/logger code a little more generic because the log message generation it isn't tied to KubeDirectorCluster.
The only downside is that I will need to pass this new logger object around in a majority of all  of the function calls. It would have been nice if I could just replace the KubeDirectorCluster parameter  in the shared/logger APIs with a new logger parameter but  the  KubeDirectorCluster parameter has to be preserved because it is still needed for the event recorder.
We should probably talk offline about possible solutions.

Reply:
I discussed the log issue with Joel yesterday offline and we decided to go ahead with the logger change. The logger setup at the beginning of each reconciliation request will be used consistently throughout the code.